pyi.Dockerfile

```diff
@@ -111,33 +111,72 @@ RUN ./node_builder_musl_env.sh
 COPY scripts/build_node_package.sh .
 RUN ./build_node_package.sh
 
-# bcc helpers
+# building bcc along with helpers
 # built on newer Ubuntu because they require new clang (newer than available in GPROFILER_BUILDER's CentOS 7)
 # these are only relevant for modern kernels, so there's no real reason to build them on CentOS 7 anyway.
-FROM ubuntu${PYPERF_BUILDER_UBUNTU} AS bcc-helpers
+FROM ubuntu${PYPERF_BUILDER_UBUNTU} AS bcc-build
 WORKDIR /tmp
 
 RUN if [ "$(uname -m)" = "aarch64" ]; then \
         exit 0; \
     fi && \
     apt-get update && \
-    apt-get install -y --no-install-recommends \
-        clang-10 \
-        libelf-dev \
-        make \
-        build-essential \
-        llvm \
-        ca-certificates \
-        git
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
+      curl \
+      build-essential \
+      iperf llvm-12-dev \
+      clang-12 libclang-12-dev \
+      cmake \
+      python3 python3-pip \
+      flex \
+      libfl-dev \
+      bison \
+      libelf-dev \
+      libz-dev \
+      liblzma-dev \
+      ca-certificates \
+      git \
+      patchelf scons
 
 COPY --from=perf-builder /bpftool /bpftool
 
 COPY scripts/bcc_helpers_build.sh .
 RUN ./bcc_helpers_build.sh
 
+# build bcc
+# TODO: copied from the main Dockerfile... but modified a lot. we'd want to share it some day.
+WORKDIR /tmp
+
+COPY ./scripts/libunwind_build.sh .
+RUN if [ "$(uname -m)" = "aarch64" ]; then \
+      exit 0; \
+    fi && \
+    ./libunwind_build.sh
+
+WORKDIR /bcc
+COPY ./scripts/pyperf_build.sh .
+RUN ./pyperf_build.sh
 
-# bcc & gprofiler
+# build staticx dedicated for PyPerf and process PyPerf binary;
+# apply patch to ensure staticx bootloader propagates dump signal to actual PyPerf binary
```

Comment:
Remind me - we chose this approach to have all different PyPerf dependency libraries packed into the final binary? What was the reason we decided not to make use of the existing `./scripts/list_needed_libs.sh` which packs relevant libraries into the staticx context of gProfiler itself?

Reply:
That's because bcc is now built with Ubuntu, borrowing process from gProfiler container build.  

Could no longer build newer version in Centos 7, due to lack of recent builds of LLVM.

Reply:
Thanks. Let's add it in a comment in this file? (we're using staticx in a separate env due to ubntuu vs centos ...)

Reply:
👍